pkg/server/secrets-encrypt.go

```diff
@@ -139,6 +140,9 @@ func encryptionEnable(ctx context.Context, server *config.Control, enable bool)
 		logrus.Infoln("Secrets encryption already disabled")
 		return nil
 	} else if providers[0].Identity != nil && providers[1].AESCBC != nil && enable {
+		if nodeArgs := getNodeArgs(server); !strings.Contains(nodeArgs, "secrets-encryption") {
+			return fmt.Errorf("secrets encryption cannot be enabled without first starting the server with --secrets-encryption flag")
```

Comment:
This won't work for RKE2, as the flag is hardcoded on in the config, but won't show up in the node-args annotation because it's not passed in via the CLI or yaml. Could we perhaps instead check for one of the secrets encryption status annotations?

Reply:
I'm removing this, it was an attempt to prevent errors discussed in https://github.com/k3s-io/k3s/issues/7195, but its unclear to me how they achieved this. You can't start 1 server with `--secrets-encryption` without all others also starting with it, its a CrititcalControlArg. You also can't run `k3s secrets-encrypt enable` when you never started with the `--secrets-encryption`, it errors out because there isn't a encryptionConfig available in the FS.